### PR TITLE
add nightly workflow

### PR DESCRIPTION
**Pull request by DematteisGiacomo**
Test on target, on self-hosted runner

### CODE REVIEW

**Comment by balaji-nordic** [comment]
Is it possible to split this into two or more reviews? For example, get the tests in one pr. And then the workflow in another?

**Comment by DematteisGiacomo** [comment]
> Is it possible to split this into two or more reviews? For example, get the tests in one pr. And then the workflow in another?

yes sure, I'll do that

File: tests/night/utils/flash_tools.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        logger.info(result.stdout)
+        logger.info("Command completed successfully.")
+    except subprocess.CalledProcessError as e:
+        # Handle errors in the command execution
+        logger.info("An error occurred while resetting the device.")
+        logger.info("Error output:")
+        logger.info(e.stderr)
+
+
+if __name__ == "__main__":
```
Comment (jorgenmk) [inline tests/night/utils/flash_tools.py:49]: Can be removed

File: tests/night/tests/test.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+HEX_FILE = os.path.abspath("tests/night/artifacts/merged.hex")
+
+def test_program_board_and_check_uart(t91x_board):
+    logger.info("PLUTO")
+    # reset_device()
+    flash_device(HEX_FILE)
+
+    expected_lines = ["Network connectivity established", "Connected to Cloud"]
+    # expected_lines = ["Network connectivity established"]
```
Comment (jorgenmk) [inline tests/night/tests/test.py:21]: Can be removed

File: tests/night/tests/test.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+logger = get_logger()
+
+TEST_TIMEOUT = 1 * 60
+
+HEX_FILE = os.path.abspath("tests/night/artifacts/merged.hex")
+
+def test_program_board_and_check_uart(t91x_board):
+    logger.info("PLUTO")
+    # reset_device()
```
Comment (jorgenmk) [inline tests/night/tests/test.py:17]: Can be removed

**Comment by DematteisGiacomo** [comment]
test execution removed, will be added in a separate PR: https://github.com/hello-nrfcloud/firmware/pull/107

**Review by jorgenmk — APPROVED** [review]
LGTM!

**Pull request merged**